pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -174,3 +202,22 @@ func (pc *PodCache) getPod(name string, namespace string) *v1.Pod {
 	}
 	return pod
 }
+
```

Comment:
Would you mind making this public ? We have quite a few conversion libraries spread across, they are quite reusable ( see my PR and doc on MCP-over-XDS as use case )